sweep/store_mock.go

```diff
@@ -2,54 +2,58 @@ package sweep
 
 import (
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/stretchr/testify/mock"
 )
 
 // MockSweeperStore is a mock implementation of sweeper store. This type is
 // exported, because it is currently used in nursery tests too.
 type MockSweeperStore struct {
-	ourTxes map[chainhash.Hash]struct{}
+	mock.Mock
 }
 
 // NewMockSweeperStore returns a new instance.
 func NewMockSweeperStore() *MockSweeperStore {
-	return &MockSweeperStore{
-		ourTxes: make(map[chainhash.Hash]struct{}),
-	}
+	return &MockSweeperStore{}
 }
 
-// IsOurTx determines whether a tx is published by us, based on its
-// hash.
+// IsOurTx determines whether a tx is published by us, based on its hash.
 func (s *MockSweeperStore) IsOurTx(hash chainhash.Hash) (bool, error) {
-	_, ok := s.ourTxes[hash]
-	return ok, nil
+	args := s.Called(hash)
+
+	return args.Bool(0), args.Error(1)
 }
 
 // StoreTx stores a tx we are about to publish.
 func (s *MockSweeperStore) StoreTx(tr *TxRecord) error {
-	s.ourTxes[tr.Txid] = struct{}{}
-
-	return nil
+	args := s.Called(tr)
+	return args.Error(0)
 }
 
 // ListSweeps lists all the sweeps we have successfully published.
 func (s *MockSweeperStore) ListSweeps() ([]chainhash.Hash, error) {
-	var txns []chainhash.Hash
-	for tx := range s.ourTxes {
-		txns = append(txns, tx)
-	}
+	args := s.Called()
 
-	return txns, nil
+	return args.Get(0).([]chainhash.Hash), args.Error(1)
 }
 
 // GetTx queries the database to find the tx that matches the given txid.
 // Returns ErrTxNotFound if it cannot be found.
 func (s *MockSweeperStore) GetTx(hash chainhash.Hash) (*TxRecord, error) {
-	return nil, ErrTxNotFound
+	args := s.Called(hash)
+
+	tr := args.Get(0)
+	if tr != nil {
+		return args.Get(0).(*TxRecord), args.Error(1)
+	}
+
+	return nil, args.Error(1)
 }
 
 // DeleteTx removes the given tx from db.
 func (s *MockSweeperStore) DeleteTx(txid chainhash.Hash) error {
-	return nil
+	args := s.Called(txid)
```

Comment:
Do these tests pass as is? Given there were no expectations declared, so in my exp, the tests fails as it declares that it got an unexpected call.

Reply:
yeah exactly, I was confused too, then realized none of the sweeper store's methods were hit in the sweeper's unit tests...they are properly mocked in the following PRs tho.